File: app/src/main/java/com/slapshotapps/dragonshockey/models/GameUpdateKeys.java
package com.slapshotapps.dragonshockey.models;

import com.slapshotapps.dragonshockey.observables.AdminObserver;

/**
 * Created on 10/30/16.
 */

public class GameUpdateKeys {

    String gameResultKey;
    String gameKey;
    String gameStatsKey;

    public GameUpdateKeys(String gameKey, String gameResultKey, String gameStatsKey) {
        this.gameKey = gameKey;
        this.gameResultKey = gameResultKey;
        this.gameStatsKey = gameStatsKey;
    }

    public String getGameResultKey() {
        return String.valueOf(gameResultKey);
    }

    public String getGameKey() {
        return String.valueOf(gameKey);
    }

    public String getGameStatsKey() {
        return String.valueOf(gameStatsKey);
    }

    public boolean gameResultKeyValid() {
        return gameResultKey != null;
    }

    public boolean gameKeyValid() {
        return gameKey != null;
    }

    public boolean gameStatsKeyValid() {
        return gameStatsKey != null;
    }
}

File: app/src/main/java/com/slapshotapps/dragonshockey/models/GameUpdateKeys.kt
package com.slapshotapps.dragonshockey.models

import androidx.annotation.Keep

/**
 * Created on 10/30/16.
 */
@Keep
class GameUpdateKeys(private val gameKey: String?, private val gameResultKey: String?,
                     private val gameStatsKey: String?) {

    fun getGameResultKey(): String {
        return gameResultKey.toString()
    }

    fun getGameKey(): String {
        return gameKey.toString()
    }

    fun getGameStatsKey(): String {
        return gameStatsKey.toString()
    }

    fun gameResultKeyValid(): Boolean {
        return gameResultKey != null
    }

    fun gameKeyValid(): Boolean {
        return gameKey != null
    }

    fun gameStatsKeyValid(): Boolean {
        return gameStatsKey != null
    }
}
